File: 2023-12-26/BT_1.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt';
const [N, parts, M, finds] = fs.readFileSync(filePath).toString().trim().split('\n');

const solution = (n, partList, m, findList) => {
  partList.sort((a, b) => a - b);
  const answer = [];

  findList.forEach((find) => {
    let isExist = false;
    let [low, high] = [0, n];

    while (low < high) {
      let mid = ~~((low + high) / 2);
      if (partList[mid] == find) {
        isExist = true;
        break;
      } else if (partList[mid] < find) {
        low = mid + 1;
      } else {
        high = mid;
      }
    }
    answer.push(isExist ? 'yes' : 'no');
  });
  console.log(answer.join(' '));
};

solution(Number(N), parts.split(' ').map(Number), Number(M), finds.split(' ').map(Number));

File: 2023-12-26/BT_1.py
n, partList, m, findList = int(input()), sorted(list(map(int, input().split()))), int(input()), list(map(int, input().split()))
answer = []

for find in findList:
    isExist = False
    low, high = 0, n

    while low < high:
        mid = (low + high) // 2
        print(mid)
        if partList[mid] == find:
            isExist = True
            break
        elif partList[mid] < find:
            low = mid + 1
        else:
            high = mid
    answer.append('yes' if isExist else 'no')

print(' '.join(answer))